ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TopicInfoCache.java

```diff
@@ -141,6 +153,74 @@ private TopicInfo load(final String topicName) {
     }
   }
 
+  private static LogicalSchema getInternalSchema(
+      final String topicName,
+      final PersistentQueryMetadata query
+  ) {
+    if (LEFT_INTERNAL.matcher(topicName).matches()) {
+      return getSchemaFromQuery(PlanSummary.PROJECT_STEP, "PrependAliasLeft", query);
+    }
+
+    if (RIGHT_INTERNAL.matcher(topicName).matches()) {
+      return getSchemaFromQuery(PlanSummary.PROJECT_STEP, "PrependAliasRight", query);
+    }
+
+    if (topicName.endsWith("Aggregate-Aggregate-Materialize-changelog")) {
+      return getSchemaFromQuery(PlanSummary.AGGREGATE_STEP, "Aggregate\\.Aggregate", query);
+    }
+
+    if (topicName.endsWith("Aggregate-GroupBy-repartition")) {
+      return getSchemaFromQuery(PlanSummary.GROUP_BY_STEP, "Aggregate\\.GroupBy", query);
+    }
+
+    if (topicName.endsWith("Reduce-changelog")) {
+      return getSchemaFromQuery(PlanSummary.SOURCE_STEP, "KsqlTopic\\.Source", query);
+    }
+
+    throw new UnsupportedOperationException("Unsupported internal topics type: " + topicName);
+  }
+
+  private static LogicalSchema getSchemaFromQuery(
+      final String stepType,
+      final String loggerSuffix,
+      final PersistentQueryMetadata query
+  ) {
+    final String executionPlan = query.getExecutionPlan();
```

Comment:
rather than getting the schema out of the plan, why not use the map from context to schema (`getSchemasDescription`)? You'd have to iterate over it to find the right suffix, but it would be just a bit less hacky.